File: backend/src/main/java/com/esun/demo/service/CommentService.java
package com.esun.demo.service;

import com.esun.demo.dto.CommentResponse;
import com.esun.demo.repository.CommentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import com.esun.demo.entity.Comment;
import java.util.List;

@Service
public class CommentService {

    @Autowired
    private CommentRepository commentRepository;

    @Transactional
    public void addComment(Long postId, String phone, String content) {
        commentRepository.createComment(postId, phone, content);
    }

    @Transactional
    public void editComment(Long commentId, String phone, String content) {
        commentRepository.updateComment(commentId, phone, content);
    }

    @Transactional
    public void removeComment(Long commentId, String phone) {
        commentRepository.deleteComment(commentId, phone);
    }

    public List<CommentResponse> getAllComments() {

        List<Comment> results = commentRepository.findAllCommentsNative();

        return results.stream().map(comment -> new CommentResponse(
                comment.getCommentId(),
                comment.getPost().getPostId(),
                comment.getUser().getUserId(),
                comment.getContent(),
                comment.getCreatedAt())).toList();
    }
}

File: frontend/src/services/comment.service.ts
import api from "../api";
import type { CreateCommentRequest, CommentResponse } from "../types/comment";

export const CommentService = {
  /**
   * 取得特定貼文的所有留言
   * @param postId 貼文 ID
   */
  async getAllComments(): Promise<CommentResponse[]> {
    const res = await api.get<CommentResponse[]>(`/comments`);
    return res.data;
  },

  /**
   * 在特定貼文下新增留言
   * @param postId 貼文 ID
   * @param data 留言內容
   */
  async addComment(
    postId: number,
    data: CreateCommentRequest,
  ): Promise<CommentResponse> {
    const res = await api.post<CommentResponse>(`/comments/${postId}`, data);
    return res.data;
  },

  /**
   * 刪除留言
   * @param commentId 留言 ID
   */
  async deleteComment(commentId: number): Promise<void> {
    await api.delete(`/comments/${commentId}`);
  },
};
